config/karma.saucelabs.js

```diff
@@ -16,42 +16,96 @@
  */
 
 const argv = require('yargs').argv;
-const glob = require('glob');
-const karma = require('karma');
 const path = require('path');
 const karmaBase = require('./karma.base');
 
+// karma.conf.js test configuration file to run.
+const testFile = argv['testfile'];
+if (!testFile) {
+  console.error('No test file path provided.');
+  process.exit(1);
+}
+
+/**
+ * Custom SauceLabs Launchers
+ */
+const browserMap = {
+  // Desktop
+  Chrome_Windows: seleniumLauncher('chrome', 'Windows 10', 'latest'),
+  Firefox_Windows: seleniumLauncher('firefox', 'Windows 10', 'latest'),
+  Safari_macOS: seleniumLauncher('safari', 'macOS 10.13', 'latest'),
+  Edge_Windows: seleniumLauncher('MicrosoftEdge', 'Windows 10', 'latest'),
+  IE_Windows: seleniumLauncher('internet explorer', 'Windows 10', 'latest')
+
+  // Mobile
+  // Safari_iOS: appiumLauncher('Safari', 'iPhone Simulator', 'iOS', '11.2'),
+  // Chrome_Android: appiumLauncher('Chrome', 'Android Emulator', 'Android', '6.0')
+};
+
+/**
+ * Any special options per package.
+ */
+const packageConfigs = {
+  messaging: {
+    // Messaging only supports these two browsers.
+    browsers: ['Chrome_Windows', 'Firefox_Windows']
```

Comment:
Can we add Edge as well (as long as it's Edge 17+)? I don't know if the tests will pass, it's been a while since I checked Edge, but they should (in theory) pass.

https://caniuse.com/#feat=push-api

Reply:
They did pass actually! But Edge wasn't listed as supported in [ENVIRONMENTS.md](ENVIRONMENTS.md) so I left it out.  Should I add it to the ENVIRONMENTS.md table too?

Reply:
We added it to the new environments doc, but I'm not sure if that's public yet. I'm guessing that we'll replace that file with a link to the doc, so no need to change it for now IMO.

Reply:
Added Edge to the messaging browser tests list.